src/Stack/Types/Config.hs

```diff
@@ -840,6 +842,8 @@ data ConfigMonoid =
     -- ^ See: 'configExtraIncludeDirs'
     ,configMonoidExtraLibDirs        :: !(Set Text)
     -- ^ See: 'configExtraLibDirs'
+    , configMonoidOverrideGccPath    :: !(First Text)
```

Comment:
Personally, I'd prefer a `First (Path Abs File)` here.

The main advantage I'd expect from that, would be better error messages than those that we currently relay from `cabal` or `ghc`.

I also think that this should be done "categorically", so for `configMonoidExtraIncludeDirs` and `configMonoidExtraLibDirs` too.

`Options.Applicative.Builder.eitherReader` would probably be a useful helper function for this.